src/Dapper.AOT.Analyzers/CodeAnalysis/DapperInterceptorGenerator.Single.cs

```diff
@@ -0,0 +1,157 @@
+﻿using Dapper.Internal;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Data;
+using System.Data.Common;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Dapper.CodeAnalysis;
+
+public sealed partial class DapperInterceptorGenerator
+{
+    void WriteSingleImplementation(
+        CodeWriter sb,
+        IMethodSymbol? method,
+        ITypeSymbol? resultType,
+        OperationFlags flags,
+        OperationFlags commandTypeMode,
+        ITypeSymbol? parameterType,
+        ImmutableArray<IParameterSymbol> methodParameters,
+        IDictionary<ITypeSymbol, int> parameterTypes,
+        IDictionary<ITypeSymbol, int> resultTypes)
+    {
+        sb.Append("return ");
+        if (HasAll(flags, OperationFlags.Async | OperationFlags.Query | OperationFlags.Buffered))
+        {
+            sb.Append("global::Dapper.DapperAotExtensions.AsEnumerableAsync(").Indent(false).NewLine();
+        }
+        // (DbConnection connection, DbTransaction? transaction, string sql, TArgs args, CommandType commandType, int timeout, CommandFactory<TArgs>? commandFactory)
+        sb.Append("global::Dapper.DapperAotExtensions.Command<");
+        if (parameterType is null || parameterType.IsAnonymousType)
+        {
+            sb.Append("object?");
+        }
+        else
+        {
+            sb.Append(parameterType);
+        }
+        sb.Append(">(cnn, ").Append(Forward(methodParameters, "transaction")).Append(", sql, ");
+        if (parameterType is null || parameterType.IsAnonymousType)
+        {
+            sb.Append("param");
+        }
+        else
+        {
+            sb.Append("(").Append(parameterType).Append(")param");
+        }
+        sb.Append(", ");
+        if (commandTypeMode == 0)
+        {   // not hard-coded
+            sb.Append(Forward(methodParameters, "commandType")).Append(HasParam(methodParameters, "commandType") ? ".GetValueOrDefault()" : "");
+        }
+        else
+        {
+            sb.Append("global::System.Data.CommandType.").Append(commandTypeMode.ToString());
+        }
+        sb.Append(", ").Append(Forward(methodParameters, "commandTimeout")).Append(HasParam(methodParameters, "commandTimeout") ? " ?? -1" : "").Append(", ");
+        if (HasAny(flags, OperationFlags.HasParameters))
+        {
+            if (!parameterTypes.TryGetValue(parameterType!, out var parameterTypeIndex))
+            {
+                parameterTypes.Add(parameterType!, parameterTypeIndex = parameterTypes.Count);
```

Comment:
note: I moved the "single" implementation, however fixed the index calculation here. It was before
```
parameterTypes.Add(parameterType!, parameterTypeIndex = resultTypes.Count);
```
and I changed to 
```
parameterTypes.Add(parameterType!, parameterTypeIndex = parameterTypes.Count);
```

it is reflected in [Query.output.cs](https://github.com/DapperLib/DapperAOT/pull/18/files#diff-8118441490fc8ef370edd4544a08f9766a7c600f50c48e6063996bed8bd54587) change

Reply:
if that is wrong - let me know and I will revert it